storage/redis_cluster.go

```diff
@@ -209,11 +220,15 @@ func (r RedisCluster) ensureConnection() {
 }
 
 // GetKey will retrieve a key from the database
-func (r RedisCluster) GetKey(keyName string) (string, error) {
+func (r *RedisCluster) GetKey(keyName string) (string, error) {
 	r.ensureConnection()
 	log.Debug("[STORE] Getting WAS: ", keyName)
 	log.Debug("[STORE] Getting: ", r.fixKey(keyName))
+	currentMu := r.getSingletonMU()
+	currentMu.RLock() //Lock rediscluster.RedisCluster.SingleRedisMode from writes
+	defer currentMu.RUnlock()
 	value, err := redis.String(r.singleton().Do("GET", r.fixKey(keyName)))
+
```

Comment:
this looks confusing - how about all other methods where we do `r.singleton().Do`, there is no any mutex there